Dashboard PCB :anguished:

Mikail-Sadic-UT opened the pull request:
# Quality Assurance Checklist
To make reviews more efficient, please make sure the board meets the following standards and check everything off once the board meets the quality check. Once everything has been checked, the assigned reviewers will begin the review process. _Edit this description to check off the list._

There are exceptions with all guidelines. As long as your decisions are justified, then you are good! Contact the reviewers or the leads about any exceptions.

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## Minimum Prerequisites
- [x] The board passes ERC and DRC.
    - Note some ERC warnings are acceptable (**_CAUTION_** please be careful)
    - Note some DRC warnings are acceptable (less so than ERC: **_CAUTION_** please be careful).
- [x] All traces are routed.
- [x] Units are in metric.
- [x] Gerbers are in a zip file in the main directory (after physical design is done)

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## Schematic Level Requirements 
- [x] Is proper noise resistance given to all peripheral devices (bypass caps and coils/ferrites)?  
- [ ] Is proper ESD protection given to all MCU input pins (zener diodes)? You could also write the input pin rating for the MCU on the schematic.  
- [ ] Is proper power protection given to peripheral devices (zener diodes)?  
- [x] Are peripheral units used properly (reading datasheet)?  
- [x] Are testing points added at useful places?  
- [ ] Is there proper short to GND protection at MCU outputs (inline resistors)?  
- [ ] Do ADC inputs have caps (appropriately sized to prevent RC delay)?  
- [ ] Are ADC inputs biased so there is room above expected value to determine if value is being overflowed?  
- [x] Are LEDs located at useful places (comm, power, debugging, extra GPIO)? They should be placed in locations   
- [x] Are parts chosen easy to collect?  
- [x] Are parts chosen easy to solder?  
- [ ] Is there reverse polarity protection on inputs?  
- [x] Do parts in the schematic match the actual part datasheet?

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## BOM Requirements
- [x] **Bill Of Materials**: is there an Excel Sheet BOM and a Mouser Cart link?
- [x] Are *all* the correct parts in the BOM?
- [x] Is each part in the Mouser Cart in stock (or do we have the part in stock)?
- [x] Does each part in the Mouser Cart have a margin for part quantity?
- [x] Does every component in the schematic have a footprint, mouser link/datasheet, and reference number associated with it?

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## Layout Level Requirements 
### 2D Spacing
- [x] The components are spaced out at an optimal distance.
    - All components can be easily hand-soldered.
    - IPC-SM-782A Standard requires a minimum distance of 1.0mm from edge to edge.
- [x] Components that are in parallel with each other are spaced out at an equal distance when possible.
- [x] The components are aligned with each other when possible.
- [x] Components are grouped based off of functionality.
    - E.g. all components for CAN should be grouped.
- [x] Bypass capacitors are less than 1cm away from their respective IC's power pins.

### 3D Spacing
- [x] Layout of components have been placed with mounting location and usage of the board in mind.
    - Are PCBs going to be stacked on above/below this board? Are tall components placed accordingly?
    - Are buttons and lights easily accessible and viewable?
    - When mounted into the car, are the heights of the components and connectors accounted for?
- [x] I am not reading every word and I need to start this checklist over.
    - Please uncheck everything you have checked and start over.
- [x] Location of connectors and wires going out of the board will prevent messy wiring.
    - Are all the wires that are going in the same direction placed on the same edge of the board?

### Components
- [x] Do footprints match the schematic?
    - Make sure pins on the footprint correspond to pins in the schematic 
    - (e.g. counterclockwise pins on footprint and alternating pins on schematic is *bad*)
- [x] Custom footprints have been double checked with the datasheet.
- [x] Footprints are consistent across the PCB: *i.e. all 0805 for Resistors/Capacitors*
- [x] Pin 1 of the footprint is labeled in some way. 
- [x] Are LED's in easy to see places? 
- [x] Are test points in easy to reach places?  
- [ ] Are critical paths of switching converters as small as possible?  

### Copper Layer 
- [x] The trace widths and trace clearances are greater than JLCPCB's minimum requirements (https://jlcpcb.com/capabilities/Capabilities). 
    - [x] Are signal traces are .15mm unless provided with reasoning?   
        - NOTE: One net can have multiple trace widths  
- [x] The trace lengths are as short as possible. 
    - Can there be a more optimal route if you go to another layer? 
- [x] Each trace's width is capable of handling the expected current flow. 
    - Use PCB width calculator to calculate trace width. 
- [x] *No sharp corners. No trace angles equal to or less than 90 degrees. 
    - Orthogonal traces should have vias if necessary. 
- [x] Are edges of board surrounded by clean ground on both layers with stitching vias?  
- [x] Traces are in parallel with each other when possible (exceptions can be made to prevent signal coupling). 
    - E.g. traces connected between an IC and MCU can be placed in parallel with each other. 
- [x] There are no random trace appendages. 
- [x] Vias placed in copper pads are fully encompassed in the copper pads. 
- [x] Through-hole components do not have extraneous vias. 
- [x] Vias should be appropriately sized according to the connected net(s) and expected current capacity.

*Not really a problem for modern manufacturing techniques but good practice and important for high speed signal integrity.

### Silkscreen Layer
- [x] Visible text sizes are greater than .8mm(L), .8mm(H), .15mm(W).
- [x] All visible text is on the silkscreen layer.
- [x] All reference labels of each component are not overlapping a copper pad or another component.
- [x] All connector pins are labeled with a meaningful and helpful name.
- [x] All LEDs are labeled with a meaningful and helpful name.
- [x] Silkscreens are mostly facing one direction (or 2). 

### Edge Cut Layer
- [x] The dimensions of the board and the mounting holes are nice values in metric i.e. no long decimals.
- [x] The physical outline of the board is on the edge cut layer.
- [x] Edge cuts are straight and parallel with opposite edge of the board.
- [x] Mounting holes are aligned and support M3 screws. All mounting holes should have distances between them shown in comments layer.
- [x] Corners are curved and contoured to mounting holes.

**Please *read* every word on *every* bullet point before checking off the corresponding box.**

## IMPORTANT NOTICE
- [x] I am confident that this board will be safe to use in a safety critical environment.
- [x] I have read every word on every bullet point and have only checked off a bullet point if I have read every word.
- [x] ((( **OPTIONAL** ))) I had fun :^)

## Other Comments
_Write any comments about the board that would help the reviewers here._
Cart:
https://www.mouser.com/ProjectManager/ProjectDetail.aspx?AccessID=b0e7218011
Gear switch: https://www.amazon.com/Yohii-Position-Vertical-Switch-SS13F11/dp/B07DNY27QB/ref=sr_1_1?crid=1IVIIMRQI2HJS&dib=eyJ2IjoiMSJ9.eq4ebiQlMkKEDFRAdhDxYwMIiYvkEjVZD_9xjgnfgCxXy-tZctG5FY5V9etNlcsO9SMg3hhRud0UZ8EvzB1QN5VE4UDG5s6WWWARvN8WRjMmnUkjLekKzcGdvc8ia7Sv9FLgmE0_jsopAeiuapqWnA.1wpn46VPwZhyZ5ITniFuaUG9biDjYCx3dxWY7euUJM8&dib_tag=se&keywords=SS13F11&qid=1732484796&sprefix=ss13f11%2Caps%2C110&sr=8-1

Mikail-Sadic-UT committed:
Dashboard PCB :anguished:

Mikail-Sadic-UT committed:
Ready for review I think :+1:

Switched design from modular to big board :). Plz rev schem thaks :P

Mikail-Sadic-UT committed:
Hotfix

Changed ignition switch to not have ON, and added supp pwr switch

Mikail-Sadic-UT committed:
Added dashboard fan connection

Not sure if i need to do anything specific with the fan (flyback diode?)

Mikail-Sadic-UT committed:
Updated dashboard

Took off some unnecessary connections

Mikail-Sadic-UT committed:
Major redisign

Added a lot of stuff and redesigned a lot of stuff

Mikail-Sadic-UT committed:
ddelet img

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/e0f6ff11-ccc4-4a56-8c81-deb28d43f346)
I would reassign all of the reference names for your components to just be the default "J5" or "U5" or whatever KiCAD automatically assigns them. Oftentimes it's easier to reference the id it gives them rather than create our own, plus the value of the connector can be the more descriptive name of what it is. Something like this:
![image](https://github.com/user-attachments/assets/e593a53c-06fd-48de-b5f1-6411a4493082)
I also think the all caps is a bit much but that's kind of a nitpick so you don't need to change that.

It's a good exercise to pull a few other boards from the organization that have been worked on recently and just check out how people typically structure/layout things.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/1884e895-93bb-40ed-ab56-f247500d2d29)
I'd like to see values and part numbers on all of these components before you start on layout at all. I'll tell you what I told people on their starter projects:
> I would put a part number on this LED, our [Standardized Mouser Components](https://utexas.sharepoint.com/:x:/r/sites/ENGR-LonghornRacing/_layouts/15/Doc.aspx?sourcedoc=%7BDDADD73C-3CE1-45DD-A622-EE3EAEDD15A2%7D&file=Standardized%20Mouser%20Components.xlsx&action=default&mobileredirect=true) should have a bunch of commonly used parts and you can find the LEDs we typically buy there. Then you can select the LED, hit E to edit the properties, and add the part number and possibly datasheet link if you really want. Once you've selected an LED, make sure that you will not blow it up (https://www.sparkfun.com/tutorials/219). Your current limiting resistor value right now seems too low.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/626d9cd2-3f75-4eb3-b2c2-318a2f1c8b07)
Is this 555 timer meant to be for blinking the lights? If so, do you think this board should blink the lights or do you think this board should just send a signal to tell the lighting board to blink the lights? I'm not completely aware of your plans to change daybreak so some context here would be appreciated 👍🏾

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/9a630ecf-2643-404d-9d69-1e7e20385d93)
Another nitpick; assuming that this would be used for heartbeat I would call this heartbeat.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/4cef4dca-dd6c-4e9a-bc3d-801c406a3d8d)
Datasheet link on this part does not lead anywhere.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/f379668b-1149-4b9e-b80e-ddf57676fe4d)
Why use this gate driver rather than a standard MOSFET circuit (such as the one used on the fan board starter project)? Is there an advantage?

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/fc44a588-fc21-408c-985a-5f76f825e91d)
Why are the right two components LEDs but the left one is a connector? Is the connector for an off-board LED?

Additionally, as I said before you will need to figure out all of your component values before moving on to layout.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/25d28de7-2267-4e4e-9630-d4e110e57835)
Nitpick but you should align this stuff. Makes the schematic look cleaner

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/e3cefb4c-95a0-4083-ac1b-490e11063fa9)
Would also make these consistent between one another

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/b9d4a5f0-3c39-4543-a0ae-4c3af924f51f)
In terms of easier wire harnessing, have you talked to electromechanical about whether separate connectors vs one big connector is easier for them? It seems like some of these pins are meant to go to different places on the car so perhaps splitting them up would be better, but you'll need to ask electromech to get context.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/feda9e90-97b4-4127-9c3e-db0ba746b8d9)
Why you need this diode for the 555 timer? Also please select a specific part number for the diode and assign it

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/da1e04dc-5127-4cfb-925e-5d3f4f07301c)
Not sure about these capacitor values of 0.01uF, I would have some justification for these. Why that number?

https://protological.com/debounce-calaculator/

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/033c057c-4234-48af-9972-4ff8ac0e06e1)
You have two diodes on the hazard switch going to RT_OUT and LFT_OUT, but there are no diodes on the blinker switch. Why not put diodes on the blinker switch as well? Provide justification.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/a1d67524-96e1-4058-8de4-47ff2c242ba3)
Double check regulations to make sure these values are acceptable.

IshDeshpa reviewed (changes requested):
Looks good for a first draft, address requested changes and I'll get someone to provide a more in-depth review.

Mikail-Sadic-UT committed:
Update 1/2

Fixed some of the requested changes.
Still need to fix it up a bit more.

Mikail-Sadic-UT committed:
Ready for review

Controls decided to move a bunch of stuff back to leaderboard, also I think I fixed everything else.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/c63ae4c9-3afb-450e-bc72-b551628e38c8)
![image](https://github.com/user-attachments/assets/c0081c6e-1989-40b7-834d-e10be6d17093)

Is the intended behavior here to not have the LED on the dashboard blink? That's fine, just wanted clarification.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/9f9e605a-96c9-4f67-b36d-5f2df5e71d9b)
What does this mean? Also, yk you can just respond in github comments right? You don't need to put stuff like this on schematic

IshDeshpa commented:
https://asenergi.com/pdf/resistors/standard-resistor-values-e24.pdf
This should have the standard resistor values that most places will probably provide for things. I'd say you probably want to change those 5ks to 5.1ks because they'll be readily available on Mouser and such. Shouldn't be much of a difference.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/0d545bbd-c6df-41f1-86bf-ab238ac5c216)
Can you test this real quick on a breadboard? Just make sure it works as expected otherwise it'll be annoying. This is kind of a nitpick though, I'm just hesitant to put hardware debouncing on there without knowing the values are correct. However it's low priority so as long as you test & verify at some point before you order the board it shouldn't be a huge deal, you can check out the parts and see the thing on an o-scope in the embedded lab in your downtime (wouldn't spend workday time on it tbh)

same for the other buttons, should take like 5 minutes to verify that the values and circuit will work.

IshDeshpa commented:
![image](https://github.com/user-attachments/assets/ac4789af-9c6a-443e-bfa2-e1bf0074eeb9)
Nitpick but you could probably just organize this schematic a little better and make it look puuuuuurdy. I always enjoy that part of it anyways :) Use the rectangles and make some little titles!

see 
![image](https://github.com/user-attachments/assets/dc22ecab-6108-446c-9cdd-64684711cb7c)

IshDeshpa reviewed (changes requested):
Can you add #auto_saved_files# and *.lck to the .gitignore? https://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository#_ignoring

Other than the stuff I've mentioned, looks good to me to start on layout.

diyarajon commented:
![image](https://github.com/user-attachments/assets/49f96308-fa68-43af-a5ef-47791e8ba651)
This might need to change depending on the rotary switch's pinouts. Could you look for a three position rotary switch that we can use? Also might be worth looking into a 3 way toggle

IshDeshpa commented:
I did actually forget about component selection. Before you start on layout, make sure you have a part number & footprint selected for every component. For passives like resistors/capacitors/LEDs, we typically use 0805 footprints whereas for everything else you'll probably need to search for parts on Mouser or use the [Standardized Mouser Components](https://utexas.sharepoint.com/:x:/r/sites/ENGR-LonghornRacing/_layouts/15/Doc.aspx?sourcedoc={DDADD73C-3CE1-45DD-A622-EE3EAEDD15A2}&file=Standardized%20Mouser%20Components.xlsx&action=default&mobileredirect=true) list

Lakshay983 commented:
For the fan connector, I have a few questions:
- did you confirm that the fans would be in a 3 wire configuration? Last I checked all the fans for this gen were 2 wire but I may be wrong on that (ask Emech)
- If they are 3 wire, did you confirm the logic level the PWM of those fans work at and did you ensure that Controls Leader is passing you the proper input (3.3V vs 12V logic)

 I like having comments for purpose but stuff like "Lakshay told me to leave this" or "this is for emech" doesn't really tell somewhat looking at the schematic what it's for. The goal of these comments is that someone who doesn't know the design or requirements can understand what this board is. Instead of this "this is for emech" consider doing "This is for driving 3 wire fans" 

Same for the supplemental connectors, I know why you need them cuz we talked about it during workday but someone looking at the schematic in a few months won't.

Second what Ishan said on testpoints for switch stuff, it's good to have them and not use it rather than needing them and not having em.

![image](https://github.com/user-attachments/assets/34141c8c-4602-4a1f-a4d4-d6a00bf814bc) what's Ignition PWR? I thought it was motor power. 

Filtering caps are good to have

This is all working at 12V logic right? Nothing 3.3V right? ccing @KnockbackNemo 

Regarding the SPDT symbol, I see there is a middle position with nothing connected to it, why not just use a SP3T and add a no connect?

You can calculate the RC time constant of your debouncing circuit if you want to see how long your debounce circuit will last, worst case you just unsolder em and pick new values.

I like the hazard diodes, just make sure you pick a diode with a very low forward voltage. 

Technically we're making a GND loop with this board and Controls Leader but I think it's fine (ccing @ppatra126 for his thoughts).
^ NVM ignore me, no GND loop

I'd recommend adding part numbers to each symbol before you start layout, it will make making BOMs much easier.

Before you start layout assign footprints and then ask for another review (footprints are very easy to mess up and not fun to debug/change after you've done layout)

diyarajon reviewed (changes requested):
Good stuff, take a look at component selection

Mikail-Sadic-UT committed:
Made pretty

Fixed a few things up and made it pretty.
Still need to double check a few things and assign all footprints

Mikail-Sadic-UT committed:
Started assigning footprints

Still have a few missing, and not sure if im doing it all right

Mikail-Sadic-UT committed:
Added excel sheet

Mikail-Sadic-UT committed:
Added a lot footprints

Some of them are custom and need to be double checked. Also no sure if they are all correctly included

Mikail-Sadic-UT committed:
Added footprints locally

Mikail-Sadic-UT committed:
Footprint all assigned, and updated to PCB

Ready for a check in to make sure everything is looking alright

KnockbackNemo commented:
<img width="428" alt="image" src="https://github.com/user-attachments/assets/b2bb5dd4-8e28-44cc-be4a-6edef33cd1e0">
I think we probably want to switch 2 and 3 (2 - Array, 3 - Motor) since we want the array on when powering the motor, but we may not always want the motor running

KnockbackNemo commented:
![image](https://github.com/user-attachments/assets/b288bcf5-813c-4b28-94ee-b07767ba706e)
![image](https://github.com/user-attachments/assets/acb68e7f-1b16-428a-9c85-f1ea622fd23e)
Is the fan controlled by feeding PWM into its expected 12V line, and what does the 12V going in do?

KnockbackNemo commented:
Everything else looks great to me! Nice work! :D

Mikail-Sadic-UT committed:
Fixed fan and rot switch

Mikail-Sadic-UT committed:
Preliminary PCB layout

Mikail-Sadic-UT committed:
Preliminary schematic

Mikail-Sadic-UT committed:
r

Mikail-Sadic-UT committed:
r pt 2

Mikail-Sadic-UT committed:
Changed layout, added 3d models

Mikail-Sadic-UT committed:
Some extra stuff and things

Rav4s commented:
- Are you sure this resistor value is high enough? For our standard LED (APTD2012LCGCK), something closer to 22.1k would probably work better
- ![image](https://github.com/user-attachments/assets/aedcdd2d-13db-421b-860f-ac8bc4b9fef7)
- Add part numbers for everything it'll make BOM a lot easier
- ![image](https://github.com/user-attachments/assets/a66baeb9-6b4f-4ba8-a8d2-5eee4732df3a)
- Is the slide switch you selected available on Mouser/Digikey? idk if I'm just blind but I can't find it anywhere
- Try placing these filter caps closer to where they're needed (the controls export connector?)
- ![image](https://github.com/user-attachments/assets/440158a1-9219-4329-8ae7-2ab7858bbf51)
- The heartbeat LED should be on the driver-facing side right?
- ![image](https://github.com/user-attachments/assets/c87cf1e1-47be-4ab2-bf53-321e9f2264f8)
- Using pads like this for testpoints will make ur life a lot easier, just keep them on the back and we can just solder wires to them for testing (footprint is in shared libraries). Also put them closer to what you're probing so you don't have to run really long traces
- ![image](https://github.com/user-attachments/assets/5e10df2e-9df8-4b62-9c70-7f406d8266e0)
- Is there a reason u had this vertical instead of horizontal like the rest?
- ![image](https://github.com/user-attachments/assets/c13a4f54-80d3-4c16-b4d5-35d876440386)
- Try to arrange your components for easier routing, like put your CAN connectors next to each other, fan control next to the fan connector, etc. Maybe smth like this would work but feel free to change it around how you want
- ![image](https://github.com/user-attachments/assets/19df3b7b-3528-4c34-9d56-1119a757fc63)
- Otherwise everything looks good, nice job!

Mikail-Sadic-UT committed:
Fixed stuff

Mikail-Sadic-UT committed:
Routing finished :0

Mikail-Sadic-UT committed:
Routed

Mikail-Sadic-UT committed:
changed routing

Mikail-Sadic-UT committed:
More stuff

Mikail-Sadic-UT committed:
Filter move >:(

Lakshay983 commented:
Schematic:

- For the ignition switch it's controlling array and motor states right? Call one array and motor  

Layout notes:

- Make sure to run DRC on the layout, there's an unconnected item and a few warnings
- This board is beeeeggg. Did you talk to emech about sizing for this board? I assume they'd prefer a smaller board, if they wanted this size then sure but I think it's better to have a small board, esp since the connectors are so far out from everything else
- Cruz Set and Cruz En are both push buttons? I assumed cruz set would be like one of those toggle switches. For this kind of stuff(including ignition) it'd be good to have LEDs on there so you can visually see what's happening when you press smth
- The reason why you power planes is to increase current capacity, increased capacitance, and a lower impedance bath between powers which makes it good for power nets like 12V and GNDPWR. For signals making them a plane adds a shit ton of noise and hella slows the signal down. Any digital signal is just a 1 or 0 toggling up and down rlly fast, so adding capacitance to the line will slow down those 1-0 transitions and mess with logic levels. Hence you don't want to make signals like CAN-H and CAN-L planes, you want to make them short traces.
- Why BPS HZ connector so far away from the controls connector. I'd rec putting it closer to the connector.
-Since the supplemental battery connector isn't electrically connected to this board maybe put it in the corner of the board and in its place you can move some buttons and switches closer to the the controls export connector

KnockbackNemo commented:
> * Cruz Set and Cruz En are both push buttons? I assumed cruz set would be like one of those toggle switches. For this kind of stuff(including ignition) it'd be good to have LEDs on there so you can visually see what's happening when you press smth

I believe cruise set has traditionally been a push button so that we don't have a case where you exit cruise control by braking, but the toggle switch is still on. @Arnav-Surjan @jiwoop Will this still be the case in Daybreak 2025, or would a toggle switch work for the new fsm?

Lakshay983 commented:
> > * Cruz Set and Cruz En are both push buttons? I assumed cruz set would be like one of those toggle switches. For this kind of stuff(including ignition) it'd be good to have LEDs on there so you can visually see what's happening when you press smth
> 
> I believe cruise set has traditionally been a push button so that we don't have a case where you exit cruise control by braking, but the toggle switch is still on. @Arnav-Surjan @jiwoop Will this still be the case in Daybreak 2025, or would a toggle switch work for the new fsm?

I see, can y'all put an indicator for the cruise mode on the new Controls Leader so there's some form of indication if power is on

KnockbackNemo commented:
So one on dashboard and one on leader?

Lakshay983 commented:
> So one on dashboard and one on leader?

probably just on the leader since ig you also want to see if the Controls code moves to the proper state and sets that LED. If it's a push button I don't think you can do that on the dashboard unless you send the signal back up. It's also on the display right

KnockbackNemo commented:
Yes that's true, we can put it on the display for the driver. Sounds good

Lakshay983 commented:
> Yes that's true, we can put it on the display for the driver. Sounds good

If y'all have extra pins then it'd be good to have an LED on the Controls Leader in case we're having display problems

KnockbackNemo commented:
> If y'all have extra pins then it'd be good to have an LED on the Controls Leader in case we're having display problems

Yup, we can do that!

jiwoop commented:
> > * Cruz Set and Cruz En are both push buttons? I assumed cruz set would be like one of those toggle switches. For this kind of stuff(including ignition) it'd be good to have LEDs on there so you can visually see what's happening when you press smth
> 
> I believe cruise set has traditionally been a push button so that we don't have a case where you exit cruise control by braking, but the toggle switch is still on. @Arnav-Surjan @jiwoop Will this still be the case in Daybreak 2025, or would a toggle switch work for the new fsm?

[These](https://app.diagrams.net/#G1_yEBhHfJ_CH7eZtytHqsMtaKGBlM4xWr#%7B%22pageId%22%3A%223kyy422RKfQy9LxuxGX_%22%7D) are the FSMs for Daybreak 2025. If you refer to the CruiseControl tab, we currently planned exiting cruise state when break is pressed as well. We can further discuss if we want to keep this or not?

If we want toggle switches, then we shouldn't be exiting the cruise with break as that would require driver to toggle back again when breaking.

Mikail-Sadic-UT committed:
Changed size and rerouted

KnockbackNemo commented:
> [These](https://app.diagrams.net/#G1_yEBhHfJ_CH7eZtytHqsMtaKGBlM4xWr#%7B%22pageId%22%3A%223kyy422RKfQy9LxuxGX_%22%7D) are the FSMs for Daybreak 2025. If you refer to the CruiseControl tab, we currently planned exiting cruise state when break is pressed as well. We can further discuss if we want to keep this or not?
> 
> 
> 
> If we want toggle switches, then we shouldn't be exiting the cruise with break as that would require driver to toggle back again when breaking.

Makes sense. We probably want to keep it as is since exiting cruise control when braking seems pretty important. Thanks, Jiwoo!

Mikail-Sadic-UT committed:
Labled n other stuff

Mikail-Sadic-UT committed:
A bunch of stuff

Rav4s commented:
- ![image](https://github.com/user-attachments/assets/3eed4c2f-5258-48e6-984a-2ab45af13ad3)
- if you move the testpoint down like this, you can route the whole IGN_ARR trace on the front, which will help with your 12v plane
- ![image](https://github.com/user-attachments/assets/7deea4d7-43e4-4c6b-8489-5a3027d70506)
- this via is not connected to the trace lol
- ![image](https://github.com/user-attachments/assets/2c7b86bc-ad98-4da3-ad9d-11fe0b05ca36)
- try moving your traces closer together so you don't end up with islands in your gnd plane. for example^
- overall looks a lot better, keep it up!

Mikail-Sadic-UT committed:
Supp changes

Champers5000 commented:
I don't have many suggestions for this board
First, if this is for sure the layout we are using, route this trace mostly on the front until it's near the ignition switch to avoid splicing the back gnd plane. 
![image](https://github.com/user-attachments/assets/f29bdac2-707f-4484-9f9d-ac12d70f3c7b)


On the topic of placement, is it really necessary to have this exact placement?
The cruise control switches, gear switch, and indicator switch out are all quite far from the controls leader export connector they route to (J1). If we perhaps move the CAN connector and Fan connector to a different edge of the board, we could make the routing much shorter. 

![image](https://github.com/user-attachments/assets/7fc4c4fa-28f4-4e45-a448-83335cc800a6)
Also these two indicator LEDs have a trace running the entire length of the board. Why not moves these closer to the connector as well?

Mikail-Sadic-UT committed:
Update Dashboard PCB.kicad_pcb

Mikail-Sadic-UT committed:
Final changes for now

added p/n n stuff

Mikail-Sadic-UT committed:
Added gerbers

Mikail-Sadic-UT committed:
helow

Mikail-Sadic-UT committed:
Uhh
